--- a/src/devicedrv/mali/platform/sunxi/sunxi.c	2020-12-14 18:20:59.115528397 +0100
+++ b/src/devicedrv/mali/platform/sunxi/sunxi.c	2020-12-14 18:22:04.339533110 +0100
@@ -3,7 +3,7 @@
 #include <linux/clkdev.h>
 #include <linux/cma.h>
 #include <linux/delay.h>
-#include <linux/dma-contiguous.h>
+#include <linux/dma-map-ops.h>
 #include <linux/dma-mapping.h>
 #include <linux/io.h>
 #include <linux/kernel.h>
