diff --git a/arch/arm/boot/dts/sun7i-a20-bananapi.dts b/arch/arm/boot/dts/sun7i-a20-bananapi.dts
index 67c8a76..7d05b25 100644
--- a/arch/arm/boot/dts/sun7i-a20-bananapi.dts
+++ b/arch/arm/boot/dts/sun7i-a20-bananapi.dts
@@ -100,6 +100,11 @@
 	cpu-supply = <&reg_dcdc2>;
 	operating-points = <
 		/* kHz	  uV */
+               1200000  1500000
+               1152000  1500000
+               1104000  1450000
+               1056000  1450000
+               1008000  1450000
 		960000	1400000
 		912000	1400000
 		864000	1350000
@@ -212,7 +217,7 @@
 &reg_dcdc2 {
 	regulator-always-on;
 	regulator-min-microvolt = <1000000>;
-	regulator-max-microvolt = <1400000>;
+	regulator-max-microvolt = <1500000>;
 	regulator-name = "vdd-cpu";
 };
 
