diff --git a/arch/arm/boot/dts/exynos4210-i9100.dts b/arch/arm/boot/dts/exynos4210-i9100.dts
index 6d0c04d77a39..9a7cc8b679db 100644
--- a/arch/arm/boot/dts/exynos4210-i9100.dts
+++ b/arch/arm/boot/dts/exynos4210-i9100.dts
@@ -300,13 +300,13 @@
 	};
 };
 
-&gpu {
-	status = "okay";
-
-	mali-supply = <&vg3d_breg>;
-	regulator-microvolt-offset = <50000>;
-	regulator-microsecs-delay = <50>;
-};
+// &gpu {
+// 	status = "okay";
+// 
+// 	mali-supply = <&vg3d_breg>;
+// 	regulator-microvolt-offset = <50000>;
+// 	regulator-microsecs-delay = <50>;
+// };
 
 &hsotg {
 	status = "okay";
