--- drivers/gpu/arm/mali/Kconfig.org	2020-10-13 18:37:57.984996360 +0200
+++ drivers/gpu/arm/mali/Kconfig	2020-10-13 18:40:07.683375726 +0200
@@ -2,7 +2,7 @@
 	tristate "Mali-300/400/450 support"
 	depends on OF && ARM && ARMGPU
 	select DMA_SHARED_BUFFER
-	---help---
+	help
 	  This enables support for the ARM Mali-300, Mali-400, and Mali-450
 	  GPUs.
 
@@ -13,19 +13,19 @@
 	string "Mali target platform"
 	default "exynos4412"
 	depends on MALI400
-	---help---
+	help
 	  This select the target platform for the Mali driver.
 
 config MALI450
 	bool "Enable Mali-450 support"
 	depends on MALI400
-	---help---
+	help
 	  This enables support for Mali-450 specific features.
 
 config MALI400_DEBUG
 	bool "Enable debug in Mali driver"
 	depends on MALI400
-	---help---
+	help
 	  This enabled extra debug checks and messages in the Mali driver.
 
 config MALI400_PROFILING
@@ -33,14 +33,14 @@
 	depends on MALI400
 	select TRACEPOINTS
 	default y
-	---help---
+	help
 	  This enables gator profiling of Mali GPU events.
 
 config MALI400_INTERNAL_PROFILING
 	bool "Enable internal Mali profiling API"
 	depends on MALI400_PROFILING
 	default n
-	---help---
+	help
 	  This enables the internal legacy Mali profiling API.
 
 choice
@@ -84,7 +84,7 @@
 	bool "Map dma-buf attachments on attach"
 	depends on MALI400 && DMA_SHARED_BUFFER
 	default y
-	---help---
+	help
 	  This makes the Mali driver map dma-buf attachments after doing
 	  attach. If this is not set the dma-buf attachments will be mapped for
 	  every time the GPU need to access the buffer.
@@ -95,7 +95,7 @@
 	bool "Support for shared interrupts"
 	depends on MALI400
 	default n
-	---help---
+	help
 	  Adds functionality required to properly support shared interrupts.  Without this support,
 	  the device driver will fail during insmod if it detects shared interrupts.  This also
 	  works when the GPU is not using shared interrupts, but might have a slight performance
@@ -105,7 +105,7 @@
 	bool "Power up Mali PMU domains in parallel"
 	depends on MALI400
 	default n
-	---help---
+	help
 	  This makes the Mali driver power up all PMU power domains in parallel, instead of
 	  powering up domains one by one, with a slight delay in between. Powering on all power
 	  domains at the same time may cause peak currents higher than what some systems can handle.
@@ -115,7 +115,7 @@
 	bool "Make Mali driver very quiet"
 	depends on MALI400 && !MALI400_DEBUG
 	default n
-	---help---
+	help
 	  This forces the Mali driver to never print any messages.
 
 	  If unsure, say N.
