diff --git a/drivers/usb/core/message.c b/drivers/usb/core/message.c
index a1ac2f0723b0..74f5b1c48586 100644
--- a/drivers/usb/core/message.c
+++ b/drivers/usb/core/message.c
@@ -2016,6 +2016,7 @@ int usb_set_configuration(struct usb_device *dev, int configuration)
 		struct usb_interface *intf = cp->interface[i];
 
 		if (intf->dev.of_node &&
+		    of_device_is_compatible(intf->dev.of_node, NULL) &&
 		    !of_device_is_available(intf->dev.of_node)) {
 			dev_info(&dev->dev, "skipping disabled interface %d\n",
 				 intf->cur_altsetting->desc.bInterfaceNumber);
