diff --git a/arch/arm/boot/dts/exynos5250-snow-common.dtsi b/arch/arm/boot/dts/exynos5250-snow-common.dtsi
index c15ecfc4077d..ad455c9b3a32 100644
--- a/arch/arm/boot/dts/exynos5250-snow-common.dtsi
+++ b/arch/arm/boot/dts/exynos5250-snow-common.dtsi
@@ -176,7 +176,7 @@ sound {
 		samsung,i2s-controller = <&i2s0>;
 	};
 
-	usb3_vbus_reg: regulator-usb3 {
+/*	usb3_vbus_reg: regulator-usb3 {
 		compatible = "regulator-fixed";
 		regulator-name = "P5.0V_USB3CON";
 		regulator-min-microvolt = <5000000>;
@@ -185,7 +185,7 @@ usb3_vbus_reg: regulator-usb3 {
 		pinctrl-names = "default";
 		pinctrl-0 = <&usb3_vbus_en>;
 		enable-active-high;
-	};
+	}; */
 
 	fixed-rate-clocks {
 		xxti {
@@ -622,12 +637,12 @@ tps65090_irq: tps65090-irq-pins {
 		samsung,pin-drv = <EXYNOS4_PIN_DRV_LV1>;
 	};
 
-	usb3_vbus_en: usb3-vbus-en-pins {
+/*	usb3_vbus_en: usb3-vbus-en-pins {
 		samsung,pins = "gpx2-7";
 		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
 		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
 		samsung,pin-drv = <EXYNOS4_PIN_DRV_LV1>;
-	};
+	}; */
 
 	max77686_irq: max77686-irq-pins {
 		samsung,pins = "gpx3-2";
@@ -703,12 +718,12 @@ &usbdrd {
 	vdd33-supply = <&ldo12_reg>;
 };
 
-&usbdrd_dwc3 {
+/* &usbdrd_dwc3 {
 	dr_mode = "host";
 };
 
 &usbdrd_phy {
 	vbus-supply = <&usb3_vbus_reg>;
-};
+}; */
 
 #include "cros-ec-keyboard.dtsi"
