--- drivers/gpu/arm/midgard/mali_kbase_sync.h.org	2021-12-30 12:45:54.903275645 +0100
+++ drivers/gpu/arm/midgard/mali_kbase_sync.h	2021-12-30 12:46:55.853688278 +0100
@@ -37,6 +37,8 @@
 
 #include "mali_kbase.h"
 
+#include <linux/fdtable.h>
+
 /**
  * struct kbase_sync_fence_info - Information about a fence
  * @fence: Pointer to fence (type is void*, as underlaying struct can differ)
@@ -156,7 +158,7 @@
  */
 static inline void kbase_sync_fence_close_fd(int fd)
 {
-	ksys_close(fd);
+	close_fd(fd);
 }
 
 /**
